test/all.js

```diff
@@ -69,6 +73,30 @@ test('checkout query param (hyperdrive)', async t => {
   )
 })
 
+test('can add a non-ready drive with key', async function (t) {
+  const store = new Corestore(RAM.reusable())
+  const drive = new Hyperdrive(store.namespace('drive'))
```

Comment:
Is it actually needed to do reusable? And same for namespace, is it needed?

Reply:
Without the reusable the corestore gets cleaned up so the added file is lost when re-opening, and without the namespace the corestore itself gets closed when closing the drive (both are artefacts of how ram works, and the last one might be unintended. But see see https://github.com/random-access-storage/random-access-memory/pull/16/files for `reusable`

This was the easiest approach I could think of to have a non-opened hyperdrive which already contains a file

Reply:
I know about `reusable` but why is it needed, if you're just testing that the key is sync?

Reply:
Is it not good enough to just put the file after `serve.add`? Anyway it's just a test so it's fine as it is

Reply:
The namespace is needed, when removed the test errors because the corestore is closed

I could put in the file after `serve.add`, but then I'm testing a slightly different flow, where the drive is readied before the file is requested. This flow is the exact one I need downstream: I test that I can add an unready drive, and that I can access the files there

Fine with removing that part and simplifying the test though, downstream can test for that